builtin/logical/mongodb/util.go

```diff
@@ -0,0 +1,82 @@
+package mongodb
+
+import (
+	"crypto/tls"
+	"errors"
+	"net"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"gopkg.in/mgo.v2"
+)
+
+// Unfortunately, mgo doesn't support the ssl parameter in its MongoDB URI parsing logic, so we have to handle that
+// ourselves. See https://github.com/go-mgo/mgo/issues/84
+func parseMongoURI(rawUri string) (*mgo.DialInfo, error) {
+	uri, err := url.Parse(rawUri)
+	if err != nil {
+		return nil, err
+	}
+
+	info := mgo.DialInfo{
+		Addrs:    strings.Split(uri.Host, ","),
+		Database: strings.TrimPrefix(uri.Path, "/"),
+		Timeout:  10 * time.Second,
+	}
+
+	if uri.User != nil {
+		info.Username = uri.User.Username()
+		password, _ := uri.User.Password()
```

Comment:
Pedantry: you can use `info.Password, _ = uri.User.Password()` here -- note `=`, not `:=`.